lib/metadata_json_lint.rb

```diff
@@ -136,10 +142,34 @@ def validate_requirements!(requirements)
       if requirement['name'] == 'pe'
         warn :requirements, "The 'pe' requirement is no longer supported by the Forge."
       end
+
+      begin
+        puppet_req = VersionRequirement.new(requirement.fetch('version_requirement', ''))
+      rescue ArgumentError => e
+        # Raised when the version_requirement provided could not be parsed
+        error :dependencies, "Invalid 'version_requirement' field in metadata.json: #{e}"
```

Comment:
Does this raise an exception or will the execution continue?

Reply:
That should raise an error in the output (using the error method) detailing that the version is not SemVer compatible. I put this in there as it was not being checked at all and because you can't use a non-SemVer version for Puppet ranges. This is the same check as is in the `validate_dependencies!` method.

Reply:
My underlying question is actually: does it proceed with `validate_puppet_ver!(puppet_req)` (which would fail).

Reply:
Could we add a test for this? If it doesn't crash then I'm fine with it but I suspect this will crash on a `NameError: undefined local variable or method`

Reply:
@ekohl There is already a test for it found here: https://github.com/voxpupuli/metadata-json-lint/blob/master/spec/version_requirement_spec.rb#L5

Reply:
What I meant for this was https://gist.github.com/ekohl/f86924dc00197890a831d57fd6329324:

```
Failures:

  1) MetadataJsonLint.validate_requirements! with invalid requirement should not raise Exception
     Failure/Error: expect { described_class.validate_requirements!(requirements) }.not_to raise_error
     
       expected no Exception, got #<NoMethodError: undefined method `mixed_syntax?' for nil:NilClass> with backtrace:
         # ./lib/metadata_json_lint.rb:169:in `validate_puppet_ver!'
         # ./lib/metadata_json_lint.rb:155:in `block in validate_requirements!'
         # ./lib/metadata_json_lint.rb:143:in `each'
         # ./lib/metadata_json_lint.rb:143:in `validate_requirements!'
         # ./spec/metadata_json_lint_spec.rb:33:in `block (5 levels) in <top (required)>'
         # ./spec/metadata_json_lint_spec.rb:33:in `block (4 levels) in <top (required)>'
     # ./spec/metadata_json_lint_spec.rb:33:in `block (4 levels) in <top (required)>'
```

The alternative is an example metadata.json.

Looks like it can be mitigated by adding an `if puppet_req` to the line below.

Also wondering if the error here should be `:dependencies`. Looking at the rest it looks like it should be `:requirements`.

Reply:
> Also wondering if the error here should be :dependencies. Looking at the rest it looks like it should be :requirements.

Could you have a look at this?